plugins/defaultplugins/common/model/srv6/keys_agent_srv6.go

```diff
@@ -0,0 +1,125 @@
+// (without worry that key part has accidentally this delimiter because otherwise it would not be one key part)
+const EtcdKeyPathDelimiter = "/"
+
+// SID (in srv6 package) is SRv6's segment id. It is always represented as IPv6 address
+type SID = net.IP
+
+// BasePrefix returns the prefix used in ETCD to store vpp SRv6 config.
+func BasePrefix() string {
+	return basePrefix
+}
+
+// LocalSIDPrefix returns longest common prefix for all local SID keys
+func LocalSIDPrefix() string {
+	return localSIDPrefix
+}
+
+// PolicyPrefix returns longest common prefix for all policy keys
+func PolicyPrefix() string {
+	return policyPrefix
+}
+
+// IsPolicySegmentPrefix check whether key has policy segment prefix
+func IsPolicySegmentPrefix(key string) bool {
+	return policySegmentPrefixRegExp.MatchString(key)
+}
+
+// SteeringPrefix returns longest common prefix for all steering keys
+func SteeringPrefix() string {
+	return steeringPrefix
+}
+
+// ParseLocalSIDKey parses SID from a key.
+func ParseLocalSIDKey(key string) (SID, error) {
+	sidStr, err := parseOneValuedKey(key, LocalSIDPrefix(), "sid")
+	if err != nil {
+		return nil, err
+	}
+	return parseIPv6(sidStr)
+}
+
+// ParsePolicyKey parses BSID from a key.
+func ParsePolicyKey(key string) (net.IP, error) {
+	sidStr, err := parseOneValuedKey(key, PolicyPrefix(), "bsid")
+	if err != nil {
+		return nil, err
+	}
+	return parseIPv6(sidStr)
+}
+
+// ParsePolicySegmentKey parses BSID of policy where policy segment belongs to and name of policy segment.
+func ParsePolicySegmentKey(key string) (net.IP, string, error) {
+	if !policySegmentPrefixRegExp.MatchString(key) {
+		return nil, "", fmt.Errorf("key %v is not policy segment key", key)
+	}
+	suffix := strings.TrimPrefix(key, policyPrefix)
+	keyComponents := strings.Split(suffix, EtcdKeyPathDelimiter)
+	if len(keyComponents) != 3 {
+		return nil, "", fmt.Errorf("key \"%v\" should have policy BSID and policy segment name", key)
+	}
+	bsid, err := parseIPv6(keyComponents[0])
+	if err != nil {
+		return nil, "", fmt.Errorf("can't parse \"%v\" into SRv6 BSID (IPv6 address)", keyComponents[0])
+	}
+	return bsid, keyComponents[2], nil
+}
+
+func parseOneValuedKey(key string, prefix string, valueName string) (value string, err error) {
+	if !strings.HasPrefix(key, prefix) {
+		return "", fmt.Errorf("key \"%v\" should have prefix \"%v\"", key, prefix)
+	}
+	suffix := strings.TrimPrefix(key, prefix)
+	keyComponents := strings.Split(suffix, EtcdKeyPathDelimiter)
+	if len(keyComponents) != 1 {
+		return "", fmt.Errorf("key \"%v\" should have %v (and only %v) after \"%v\"", key, valueName, valueName, prefix)
+	}
+	return keyComponents[0], nil
+}
+
+// parseIPv6 parses string <str> to IPv6 address (including IPv4 address converted to IPv6 address)
+func parseIPv6(str string) (net.IP, error) {
+	ip := net.ParseIP(str)
+	if ip == nil {
+		return nil, fmt.Errorf("\"%v\" is not ip address", str)
```

Comment:
You can use `%q` format verb to display raw form of string with quotes.

Reply:
fixed in 6d969cac6d5d82b3af89ad8b567b301ab4def4db